ocaml/testsuite/tests/typing-modal-kinds/basics.ml

```diff
@@ -683,3 +679,130 @@ let foo : (string -> string) -> (string -> string) @ unique
 [%%expect{|
 val foo : (string -> string) -> unique_ (string -> string) = <fun>
 |}]
+
+(* CR layouts: maybe use the helper function below to refactor the tests above,
+   as well as [expected_mode.ml]. *)
+
+let weaken_immutable_data : 'a -> 'a @ contended once nonportable =
```

Comment:
Perhaps add a test or two that fails (e.g. on `(int -> int) ref`) just to validate the test structure.